Library/Homebrew/cask/lib/hbc/installer.rb

```diff
@@ -76,20 +77,42 @@ def stage
     def install
       odebug "Hbc::Installer#install"
 
-      if @cask.installed? && !force
+      if @cask.installed? && !force && !@reinstall
         raise CaskAlreadyInstalledAutoUpdatesError, @cask if @cask.auto_updates
         raise CaskAlreadyInstalledError, @cask
       end
 
       print_caveats
       fetch
+      uninstall_if_neccessary
```

Comment:
Make this `uninstall_if_neccessary if @reinstall`, and remove `@reinstall` from the `uninstall_if_neccessary` method.